ynh_add_config/ynh_add_config

```diff
@@ -0,0 +1,135 @@
+    local -A args_array=( [t]=template= [d]=destination= )
+    local template
+    local destination
+    # Manage arguments with getopts
+    ynh_handle_getopts_args "$@"
+    local template_path
+
+    if [ -f "../conf/$template" ]; then
+        template_path="../conf/$template"
+    elif [ -f "../settings/conf/$template" ]; then
+        template_path="../settings/conf/$template"
+    elif [ -f "$template" ]; then
+        template_path=$template
+    else
+        ynh_die --message="The provided template $template doesn't exist"
+    fi
+
+    ynh_backup_if_checksum_is_different --file="$destination"
+
+    cp "$template_path" "$destination"
+
+    ynh_replace_vars --file="$destination"
+
+    ynh_store_file_checksum --file="$destination"
+}
+
+# Replace variables in a file
+#
+# usage: ynh_replace_vars --file="file"
+# | arg: -f, --file=     - File where to replace variables
+#
+# The helper will replace the following keywords with global variables
+# that should be defined before calling this helper :
+#   __PATH__      by  $path_url
+#   __NAME__    by $app
+#   __NAMETOCHANGE__    by $app
+#   __USER__    by $app
+#   __FINALPATH__    by $final_path
+#   __PHPVERSION__    by $YNH_PHP_VERSION
+#
+# And any dynamic variables that should be defined before calling this helper like:
+#   __DOMAIN__    by $domain
+#   __APP__    by $app
+#   __VAR_1__    by $var_1
+#   __VAR_2__    by $var_2
+#
+#
+ynh_replace_vars () {
+    # Declare an array to define the options of this helper.
+    local legacy_args=f
+    local -A args_array=( [f]=file= )
+    local file
+    # Manage arguments with getopts
+    ynh_handle_getopts_args "$@"
+
+    # Replace specific YunoHost variables
+    if test -n "${path_url:-}"
+    then
+        # path_url_slash_less is path_url, or a blank value if path_url is only '/'
+        local path_url_slash_less=${path_url%/}
+        ynh_replace_string --match_string="__PATH__/" --replace_string="$path_url_slash_less/" --target_file="$file"
+        ynh_replace_string --match_string="__PATH__" --replace_string="$path_url" --target_file="$file"
+    fi
+    if test -n "${app:-}"; then
+        ynh_replace_string --match_string="__NAME__" --replace_string="$app" --target_file="$file"
+        ynh_replace_string --match_string="__NAMETOCHANGE__" --replace_string="$app" --target_file="$file"
+        ynh_replace_string --match_string="__USER__" --replace_string="$app" --target_file="$file"
+    fi
+    if test -n "${final_path:-}"; then
+        ynh_replace_string --match_string="__FINALPATH__" --replace_string="$final_path" --target_file="$file"
+    fi
+    if test -n "${YNH_PHP_VERSION:-}"; then
+        ynh_replace_string --match_string="__PHPVERSION__" --replace_string="$YNH_PHP_VERSION" --target_file="$file"
+    fi
+
+    # Replace othes variables
+
+    # List other unique (__ __) variables in $file
+    local uniques_vars=( $(grep -o '__[^.]*__' $file | sort --unique | sed "s@__\([^.]*\)__@\L\1@g" ))
```

Comment:
```suggestion
    local uniques_vars=( $(grep -o '__[A-Z0-9]+__' $file | sort --unique | sed "s@__\([^.]*\)__@\L\1@g" ))
```

I think I would grep for uppercase stuff here, to be consistent with the fact that later we replace `__${one_var^^}_` (i.e. uppercase version of `$one_var`. Otherwise we could have some people naively trying to have `__foo__` (in lowercase) and wondering why the helper doesn't replace it even though this grep catches it ... (well, one could argue that this new code also silently ignore it so idk)

Reply:
Thanks, I didn't though of that case :)

Reply:
After some test I had to switch the grep expression  to `__[A-Z0-9_]*__`
- adding missing `_` used for example on `__DB_USER__
- and `+` give an empty list when `*` give the good list of variables